packages/shared/auth0-client/src/index.ts

```diff
@@ -103,6 +103,18 @@ export default class Auth0Client {
     });
   }
 
+  async validateUserCredentials(username: string, password: string) : Promise<APIResponse<boolean>> {
+    return this.getInstanceWithCredentials(username, password)
+      .then(_ => successResponse(true))
+      .catch(error => {
```

Comment:
Similar to the other functions in here, I think we should be checking for specific error codes rather than just catching any error with has a response has an invalid credentials.

For example, if invalid credentials are provided then we'll get a 401 back - I think we should specifically check for this and return `InvalidCredentials`, otherwise return a generic error.

Reply:
Huh, not sure what happened here. I thought I did check for the 401. IIRC 401 is the only error that can be returned here though, so it should only be a 401 or generic unhandled error I think.

Reply:
Yeah AFAICT `401` is the only non-`200` response code that can be returned, but if we catch any error and return it as an `InvalidCredentials` back to the caller then _if_ other response codes are available / introduced, then that will start returning as an `InvalidCredentials` too.